src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -894,10 +894,13 @@ private Error GetError(JsonSchema schema, Error error)
             }
 
             if (!string.IsNullOrEmpty(error.PropertyPath) &&
-                schema.Rules.TryGetValue(error.PropertyPath, out var attributeCustomRules) &&
+                schema.Rules.TryGetValue(error.PropertyPath, out var attributeCustomRules) && // todo remove schema.Rules to CustomRuleProvider
```

Comment:
let's create a task for tracking these future improvements

Reply:
Dev Task 332944: Refactor CustomRules related code in docfx